src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.NormalPriorityProcessor.cs

```diff
@@ -376,7 +376,7 @@ private async Task ProcessDocumentAsync(ImmutableArray<IIncrementalAnalyzer> ana
                         {
                             // we got cancelled in the middle of processing the document.
                             // let's make sure newly enqueued work item has all the flag needed.
-                            if (!processedEverything)
+                            if (!processedEverything && !CancellationToken.IsCancellationRequested)
```

Comment:
LowPriorityProcessor seems to also have this check in ProcessProjectAsync.   Do we need to fix that up as well?

Reply:
Updated `LowPriorityProcessor` and `HighPriorityProcessor`, and checked other related types to see if any other cases were relevant (didn't see any).

Reply:
Ok... so i have to admit i find this code prettttty confusing.  It looks like 'processedEverything' is set to `cancellationToken.IsCancellationRequested`.  So is this check just checking if hte value changed before setting it right before the finally?

Reply:
Oh........   there are two cancellation tokens.  Ok.  Even more confusing.  That said, can we get rid of the strange control flow of processedEverything and just do:

`var processedEverything = !cancellationToken.IsCancellationRequested && !CancellationToken.IsCancellationRequested`?

It doesn't seem that helpful IMO to actually have the variable and set it based on the cancellation token above... or maybe i'm just not getting something here.